fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryStreamingAggregationPlan.java

```diff
@@ -120,37 +136,46 @@ public boolean isReverse() {
         return ImmutableList.of(inner);
     }
 
+    @Nonnull
+    @Override
+    public Set<Type> getDynamicTypes() {
+        return ImmutableSet.copyOf(Iterables.concat(
```

Comment:
If `Type`s don't change across multiple plan executions, does it make sense to amortise this?